components/playlist/browser/playlist_service.cc

```diff
@@ -422,21 +431,29 @@ base::WeakPtr<PlaylistService> PlaylistService::GetWeakPtr() {
 
 void PlaylistService::FindMediaFilesFromContents(
     content::WebContents* contents,
-    FindMediaFilesFromContentsCallback callback) {
+    FindMediaFilesFromContentsCallback callback,
+    bool allow_to_create_background_web_contents) {
   CHECK(*enabled_pref_) << "Playlist pref must be enabled";
-  DCHECK(contents);
+  CHECK(contents);
 
-  PlaylistDownloadRequestManager::Request request;
   auto current_url = contents->GetVisibleURL();
-  if (ShouldGetMediaFromBackgroundWebContents(contents)) {
-    request.url_or_contents = current_url.spec();
-  } else {
-    request.url_or_contents = contents->GetWeakPtr();
+  auto* tab_helper = PlaylistTabHelper::FromWebContents(contents);
```

Comment:
accessing the `PlaylistTabHelper` from the `PlaylistService` makes this incompatible with ios. Also the responsibilities of the classes are now unclear. Sometimes the `PlaylistService`. The fundamental issue seems to be the use of `found_items()`  in the tab helper vs just always returning them in a callback. You can cache the results in the tab helper so you don't have to run the extraction again, but the logic seems hard to follow right now.